robottelo/hosts.py

```diff
@@ -61,6 +61,20 @@ def get_sat_version():
     return Version('9999' if 'nightly' in sat_version else sat_version)
 
 
+def get_sat_rhel_version():
```

Comment:
I would like to see this cached while seeding this to the user_properties, be it call outside the loop  and use or cachec this function is fine !

Reply:
> We can actually ignore caching here since there could be instances when `settings.server.hostname` is changed and this method is used. If a cached version was used, then an incorrect version could be returned.

@JacobCallahan @jyejare Also, caching didn't make much difference in execution time so I removed the caching for now.

Reply:
@shweta83 Right, it won't because you have stored the RHEL version before you started looping over items.

@JacobCallahan Even though `settings.server.hostname` changes, there is 0% chance that test will ask to deploy RHEL8 satellite when tests running on RHEL7 pipeline.